pilot/pkg/model/validation.go

```diff
@@ -244,7 +247,12 @@ func checkDNS1123Preconditions(name string) error {
 }
 
 func validateDNS1123Labels(domain string) error {
-	for _, label := range strings.Split(domain, ".") {
+	parts := strings.Split(domain, ".")
+	topLevelDomain := parts[len(parts)-1]
+	if len(topLevelDomain) > 0 && '0' <= topLevelDomain[0] && topLevelDomain[0] <= '9' {
```

Comment:
can we use the regex on top of this file?

Reply:
Also are you sure this is per dns spec that tld cannot start with a number?

Reply:
https://tools.ietf.org/html/rfc3696#section-2

Not sure what it really means for validation though. The more important question is where in the networking model to we want to allow only DNS names and not IPs. If the answer is nowhere, then I guess the old ValidateFQDN function is fine, and we should clean up any places in the code that call it and also redundantly check for a valid IP